JumpscaleCore/servers/gedis/handlers.py

```diff
@@ -300,21 +302,35 @@ def _handle_request(self, request, address, user_session):
         elif request.command.command == "auth":
             tid, seed, signature = request.arguments
             tid = int(tid)
-            try:
-                tclient = j.clients.threebot.client_get(threebot=tid)
-            except Exception as e:
-                logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
-                return (logdict, None)
-            try:
-                verification = tclient.verify_from_threebot(seed, signature)
-            except Exception as e:
-                logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
-                return (logdict, None)
-            # if we get here we know that the user has been authenticated properly
-            user_session.threebot_id = tclient.tid
-            user_session.threebot_name = tclient.name
-            j.shell()
 
+            current_threebot_id = int(j.tools.threebot.me.default.tid)
+            # If working on same machine no need to get a client to authenticate
+            # otherwise, we'll have infinite loop
+            if current_threebot_id != tid:
+                try:
+                    tclient = j.clients.threebot.client_get(threebot=tid)
```

Comment:
For a brand new user 3bot that's still in the [registeration](https://github.com/threefoldtech/jumpscaleX_threebot/blob/development/ThreeBotPackages/tfgrid/registration/actors/registration.py#L36) phase, it will need to execute some actors on the explorer without having a tid yet, because it first registers the user which in turn creates the tid. And in this case, the tid sent to the server (explorer in this case) will be 0 and ` j.clients.threebot.client_get` asserts on the value being larger than zero otherwise it fails.